kotlinx-coroutines-core/jvm/src/ThreadContextElement.kt

```diff
@@ -111,32 +112,53 @@ public interface ThreadContextElement<S> : CoroutineContext.Element {
  *         traceThreadLocal.set(oldState)
  *     }
  *
- *     override fun copyForChildCoroutine(): CopyableThreadContextElement<TraceData?> {
+ *     override fun copyForChild(): TraceContextElement {
  *         // Copy from the ThreadLocal source of truth at child coroutine launch time. This makes
  *         // ThreadLocal writes between resumption of the parent coroutine and the launch of the
  *         // child coroutine visible to the child.
  *         return TraceContextElement(traceThreadLocal.get()?.copy())
  *     }
+ *
+ *     override fun mergeForChild(overwritingElement: CoroutineContext.Element): CoroutineContext {
+ *         // Merge operation defines how to handle situation when both
+ *         // the parent coroutine has an element in the context and
+ *         // an element with the same key was also
+ *         // explicitly passed to the child coroutine.
+ *         // If merging does not require special behavior,
+ *         // the copy of the element can be returned.
+ *         return TraceContextElement(traceThreadLocal.get()?.copy())
+ *     }
  * }
  * ```
  *
  * A coroutine using this mechanism can safely call Java code that assumes it's called using a
  * `Thread`.
  */
+@DelicateCoroutinesApi
 @ExperimentalCoroutinesApi
 public interface CopyableThreadContextElement<S> : ThreadContextElement<S> {
 
     /**
      * Returns a [CopyableThreadContextElement] to replace `this` `CopyableThreadContextElement` in the child
-     * coroutine's context that is under construction.
+     * coroutine's context that is under construction if the child coroutine's context does not contain
```

Comment:
"Child coroutine's context" still irks me: "Returns a Z to replace Y in the X that is under construction if *X* does not contain…" Taken literally, this definition doesn't converge! This probably *is* fairly unambiguous, but still.

Could we maybe evaluate some other options for ?
* "the appended part"
* "the new part of the context" 
* "the added context" (the parameter name for `CoroutineContext.newCoroutineContext`)
* Something else?

Reply:
>Returns a [CopyableThreadContextElement] to replace `this` `CopyableThreadContextElement` in the child
coroutine's context that is under construction if the inherited parent context does not contain
an element with the same [key].

WDYT?

Reply:
I think that, since `merge` is called on the context element inherited from the parent, we need to phrase that in terms of an operation on the parent context, and therefore, for consistency, here as well.

Reply:
Could you please elaborate on what do you propose?

>if the inherited parent context does not contain an element with the same [key].

For me it seems like the doc is phrased in terms of parent context

Reply:
It's not "frome the point of view" of the parent context, if that makes sense.

Some specific proposals:
```suggestion
     * coroutine's context that is under construction if the added context does not contain an element with the same key.
```
```suggestion
     * coroutine's context that is under construction if the context explicitly passed for construction does not contain an element with the same key.
```